### PR TITLE
Apply asc format since format v8.1

### PR DESCRIPTION
**Pull request by Tian-Jionglu**
A solution to solve issue #1531.

Mainly include 2 changes in `ASCReader` class:
1. extract header when initial ASCReader, to get correct `start_time`, `version`, `date`, etc before iterate the messages. I think this is necessary;
2. new function `_process_fd_can_frame_2` to process CANFD message in ASCFormat since V8.1;

Remaining issue:
some tests can not pass, mainly because after the change, correct `absolute` timestamp is get when iterate the messages;

### CODE REVIEW

**Comment by zariiii9003** [comment]
@Tian-Jionglu Are you still working on the tests? I cannot merge if the tests are failing.

**Comment by Tian-Jionglu** [comment]
Commits to parse the ASC format version using `packaging`.
Jobs in working: change ASC format test to adjust absolute time.

**Comment by Tian-Jionglu** [comment]
@hardbyte @zariiii9003 ,
In example data, 
test/data/logfile.asc
test/data/logfile_errorframes.asc
test/data/test_CanErrorFrames.asc
test/data/test_CanFdMessage.asc
test/data/test_CanFdMessage64.asc
test/data/test_CanMessage.asc
test/data/test_CanRemoteMessage.asc

the ASC format version `// version 9.0.0` is actually logged, or only written for example?

The format in these files may be before `8.1.0`.

**Review by hardbyte — APPROVED** [review]

**Comment by Tian-Jionglu** [comment]
When diving into test example, i found my mistake at the start.
The symbolic name issue is not only related to ASC format version. In fact, SymblicName works for symbolic logging at first place.
I'm still looking for example about symblic logging.

I would close this PR in coming days, and start a new PR to inherit the change to fix absolute time in ASC format.
If there is already support about symbolic logging in python-can, please give me a hint.

File: can/io/asc.py [hunk 1]
```diff
@@ -98,14 +127,26 @@ def _extract_header(self) -> None:
...
             if comment_match:
                 continue
 
             if events_match:
                 self.internal_events_logged = events_match.group("no_events") is None
-                break
+                continue
 
-            break
+            if trigger_match:
```
Comment (zariiii9003) [inline can/io/asc.py:142]: I think the files can contain multiple triggers. Parsing it as part of the header is probably not sufficient.

**Comment by Tian-Jionglu** [comment]
see PR #1788

**Pull request closed without merging**